config/crd/bases/tests.testkube.io_tests.yaml

```diff
@@ -0,0 +1,64 @@
+
+---
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.4.1
+  creationTimestamp: null
+  name: tests.tests.testkube.io
+spec:
+  group: tests.testkube.io
+  names:
+    kind: Test
+    listKind: TestList
+    plural: tests
+    singular: test
+  scope: Namespaced
+  versions:
+  - name: v1
+    schema:
+      openAPIV3Schema:
+        description: Test is the Schema for the tests API
+        properties:
+          apiVersion:
+            description: 'APIVersion defines the versioned schema of this representation
+              of an object. Servers should convert recognized schemas to the latest
+              internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+            type: string
+          kind:
+            description: 'Kind is a string value representing the REST resource this
+              object represents. Servers may infer this from the endpoint the client
+              submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+            type: string
+          metadata:
+            type: object
+          spec:
+            description: TestSpec defines the desired state of Test
+            properties:
+              scripts:
+                description: Scripts is list of scripts which will be sequentially
+                  orchestrated
+                items:
```

Comment:
we could make this a little more future-proof by letting a Test consist of "Steps" - where each step can be of a specific type; for now we would only support "executeScript" steps - which would have a reference to the script to be executed  and a list of parameters to pass to the script. Future step types will be "delay", "executeTest" (for running other tests from within a test - to allow for modularisation), "loadData", etc...

Reply:
@olensmar check recent commit, I think it could be quite flexible - I'm adding fields which configure particular step type ( added delay and script execution for now)